hw/top_earlgrey/data/ip/chip_i2c_testplan.hjson

```diff
@@ -0,0 +1,135 @@
+      stage: V2
+      si_stage: SV2
+      tests: ["chip_sw_i2c_device_tx_rx"]
+    }
+    {
+      name: chip_sw_i2c_speed
+      desc: '''Verify the transmission speed of data over the chip's I2C interfaces.
+
+            Run the following for standard, fast and fast plus speeds:
+            - Program the I2C to be in the appropriate mode (host/target).
+            - For hosts, the SW test writes a known payload over the chip's I2C host interface.
+              It should then receive back that payload plus one.
+              The test checks whether the payload is sent at the appropriate baud-rate.
+            - For targets, the test sends data at the appropriate baud-rate and checks whether an ack is received.
+              The test should also do a read request.
+
+            Verify all instances of I2C in the chip.
+
+            Notes for silicon targets:
+            - The testbench in this case will be a device that physically toggles/inspects the appropriate SCL and SDA lines.
+            - To check the baud rate the test bench should time the packets and average this time out over multiple runs.
+            Then check whether these timings match the appropriate baud-rate within an acceptable range:
+            Fast-mode should be faster than standard-mode and fast-mode plus should be faster than fast-mode.
+            '''
+      features: ["I2C.MODE.HOST", "I2C.MODE.TARGET", "I2C.SPEED.STANDARD", "I2C.SPEED.FAST", "I2C.SPEED.FASTPLUS"]
+      stage: V3
+      si_stage: SV2
+      tests: []
+    }
+    {
+      name: chip_sw_i2c_override
+      desc: '''Verify overriding SDA and SCL on I2C interfaces.
+
+            - Program the I2C to be in host mode.
+            - Software cycles through all SDA/SCL combinations with long delays.
+            - Test monitors the SDA/SCL lines and checks whether it receives the values in the expected order and delay.
+
+            Verify all instances of I2C in the chip.
+
+            Notes for silicon targets:
+            - The testbench in this case will be a device that physically toggles/inspects the appropriate SCL and SDA lines.
+            '''
+      features: ["I2C.MODE.HOST", "I2C.OVERRIDE"]
+      stage: V3
+      si_stage: SV2
+      tests: []
+    }
+    {
+      name: chip_sw_i2c_clockstretching
+      desc: '''Verify clock stretching on I2C interfaces.
+
+            For all I2C host mode instances:
+            - Program the I2C to be in host mode.
+            - Do a read and make target stretch after reading an address.
+            - Do a write and make the target stretch during that write.
+            - Do a read and make the target stretch during that read.
+
+            Notes for silicon targets:
+            - The testbench in this case will be a device that physically toggles/inspects the appropriate SCL and SDA lines.
+            '''
+      features: ["I2C.MODE.HOST", "I2C.PROTOCOL.CLOCKSTRETCHING"]
```

Comment:
[Clock stretching](https://opentitan.org/book/hw/ip/i2c/doc/theory_of_operation.html#clock-stretching) is supported in target mode as well and it would be good to have a separate test case for it.

Reply:
I specified this host test because during SiVal testing the testbench has control over the target and can do the clockstretching. How do you force the OpenTitan target to do clock stretching, even when the test bench only has control over the host?

Reply:
Agreed but I2C supports clock stretching in target mode as well (and is allowed in I2C protocol). For instance, when ACQ FIFO is full and Ibex is busy with other higher priority interrupts and delays reading from the FIFO, OT I2C target holds SCL low until ACQ FIFO has space for additional bytes. This scenario can be triggered in SiVal.

`chip_sw_i2c_clockstretching` only focuses on OT I2C host mode. As each I2C IP instance in OT is used either as a Host or a Target, , there should be a separate target mode test that covers clock stretching scenarios mentioned in https://opentitan.org/book/hw/ip/i2c/doc/theory_of_operation.html#clock-stretching. 
There is a DV test that covers some of these scenarios https://opentitan.org/book/hw/ip/i2c/data/i2c_testplan.html#target_clock_stretch

Reply:
Thank Ravi! I've made an initial attempt at this. Happy to change it though.

Reply:
Thanks Marno. This looks good for basic clock stretch in target mode. Do we need to exhaustively cover all cases of clock stretch in SiVal ?

Reply:
@crteja / @marnovandermaas 
For SiVal Phase-II we should cover base cases.
For SiVal Phase-III this is where we should cover exhaustive testing

Reply:
@marnovandermaas , would it make sense to have a separate SV3 test that covers all cases of clock stretching and mark this test as SV2 ?

Reply:
Ok, I've marked this as SV2 again.